File: dotnet/examples/ServerConfiguration/Metrics/TopicMetricCollector/RemoveTopicMetricCollector.cs
/**
 * Copyright © 2023 - 2024 Diffusion Data Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

using System;
using System.Threading;
using System.Threading.Tasks;
using PushTechnology.ClientInterface.Client.Factories;
using PushTechnology.ClientInterface.Client.Features.Metrics;
using static System.Console;
using static PushTechnology.ClientInterface.Examples.Program;

namespace PushTechnology.ClientInterface.Examples.ServerConfiguration.Metrics.TopicMetricCollector
{
    public sealed class RemoveTopicMetricCollector : Example
    {
        public override async Task Run(CancellationToken cancellationToken, string[] args)
        {
            string serverUrl = args[0];

            var session = Diffusion.Sessions
                .Principal("admin")
                .Credentials(Diffusion.Credentials.Password("password"))
                .Open(serverUrl);

            ITopicMetricCollector collector = null;
            string topicSelector = "?my/topic//";

            var builder = Diffusion.NewTopicMetricCollectorBuilder();
            builder = (ITopicMetricCollectorBuilder)builder.ExportsToPrometheus(false);
            builder = builder.GroupByTopicType(true);
            builder = builder.GroupByTopicView(true);
            builder = builder.GroupByPathPrefixParts(15);
            builder = (ITopicMetricCollectorBuilder)builder.MaximumGroups(10);
            collector = builder.Create("Topic Metric Collector 1", topicSelector);

            await session.Metrics.PutTopicMetricCollectorAsync(collector, cancellationToken);

            await session.Metrics.RemoveTopicMetricCollectorAsync(collector.Name, cancellationToken);

            WriteLine($"{collector.Name} has been removed.");

            session.Close();
        }
    }
}

File: python/examples/src/diffusion_examples/server_configuration/metrics/topic_metric_collector/remove_topic_metric_collector.py
"""
Copyright © 2024 Diffusion Data Ltd.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
"""

import asyncio

from diffusion import sessions, Credentials
from diffusion.features.control.metrics.topic_metrics import (
    TopicMetricCollectorBuilder,
)


from diffusion_examples.utils.program import Example


class RemoveTopicMetricCollector(Example):
    async def run(
        self,
        server_url: str = "<url>",
        principal: str = "<principal>",
        password: str = "<password>",
    ):
        async with sessions().principal(principal).credentials(
            Credentials(password)
        ).open(server_url) as session:
            topic_selector = "?my/topic//"
            builder = TopicMetricCollectorBuilder().export_to_prometheus(False)
            builder = builder.group_by_topic_type(True)
            builder = builder.group_by_topic_view(True)
            builder = builder.group_by_path_prefix_parts(15)
            builder = builder.maximum_groups(10)
            collector = builder.create(
                "Topic Metric Collector 1", topic_selector
            )
            await session.metrics.put_topic_metric_collector(collector)
            await session.metrics.remove_topic_metric_collector(collector.name)
            print(f"{collector.name} has been removed.")


if __name__ == "__main__":
    asyncio.run(
        RemoveTopicMetricCollector().run(
            server_url="ws://localhost:8080",
            principal="admin",
            password="password",
        )
    )
